### PR TITLE
Add filter by expression option to samtools_view

### PR DESCRIPTION
**Pull request by fubar2**
FOR CONTRIBUTOR:
* [x] This PR updates an existing tool or tool collection

- ~~Update samtools dependency version to 1.20~~ see #6285 - I don't have the skills...
- Added `filtering by an expression` using the -e command line option such as `-e sclen>10 ` to retain any read with more than 10 soft clipped nucleotides. 
- Added 2 tests including @bgruening's request to filter all reads from a contig like chrM

```
-e STR, --expr STR

    Only include alignments that match the filter expression STR. The syntax for these expressions is described in the main samtools(1) man page under the FILTER EXPRESSIONS heading. 
```

### CODE REVIEW

**Comment by fubar2** [comment]
Multiple test failures with an attempted update to 1.20 so testing reverting back to 1.15.1

**Comment by fubar2** [comment]
@bgruening: there are two samtools tests broken by the upgrade - need help to fix but this PR works fine with 1.15 so here it is - once the other draft PR is fixed, it will upgrade everything?

File: tool_collections/samtools/samtools_view/samtools_view.xml [hunk 1]
```diff
@@ -300,6 +302,22 @@
...
+                        <param name="select_expr" type="select" label="Filter by expression">
+                            <option value="no" selected="True">No</option>
+                            <option value="yes">Filter using an expression (see manual)</option>
+                        </param>
+                        <when value="no"/>
+                        <when value="yes">
+                            <param name="expression" type="text" argument="-e" label="Filter by expression - for example sclen&gt;0 will filter all soft clipped reads" help="See Samtools manual for Filter expression syntax">
+                                <sanitizer invalid_char="">
+                                    <valid initial="string.printable">
+                                    <remove value=" "/>
```
Comment (bgruening) [inline tool_collections/samtools/samtools_view/samtools_view.xml:315]:
```suggestion
                                    <remove value=" "/>
                                    <remove value="'"/>
                                    <remove value='"'/>
```

Maybe?

File: tool_collections/samtools/samtools_view/samtools_view.xml [hunk 2]
```diff
@@ -908,6 +926,50 @@
...
+        </test>
+         <!-- 33) testing expression filters -->
+        <test expect_num_outputs="1">
+            <param name="input" value="in_test_30.bam" ftype="bam"/>
+            <conditional name="mode">
+                <param name="outtype" value="selected_reads" />
+                <section name="filter_config">
+                    <conditional name="cond_expr">
+                        <param name="select_expr" value="yes"/>
+                        <param name="expression" value='rname!="chr13"'/>
```
Comment (bgruening) [inline tool_collections/samtools/samtools_view/samtools_view.xml:959]: Thanks!

File: tool_collections/samtools/samtools_view/samtools_view.xml [hunk 3]
```diff
@@ -908,6 +926,50 @@
...
+                <conditional name="output_options">
+                    <conditional name="output_format">
+                        <param name="oformat" value="bam" />
+                    </conditional>
+                </conditional>
+            </conditional>
+            <assert_command>
+                <has_text text="-e 'rname!="/>
+            </assert_command>
+            <output name="outputsam" file="test_33.bam" ftype="bam" lines_diff="2" />
```
Comment (bgruening) [inline tool_collections/samtools/samtools_view/samtools_view.xml:971]:
```suggestion
            <output name="outputsam" file="test_33.bam" ftype="bam" lines_diff="2" ftype="bam" />
```

File: tool_collections/samtools/samtools_view/samtools_view.xml [hunk 4]
```diff
@@ -908,6 +926,50 @@
...
+                <conditional name="output_options">
+                    <conditional name="output_format">
+                        <param name="oformat" value="bam" />
+                    </conditional>
+                </conditional>
+            </conditional>
+            <assert_command>
+                <has_text text="-e 'sclen>0'"/>
+            </assert_command>
+            <output name="outputsam" file="test_32.bam" ftype="bam" lines_diff="2" />
```
Comment (bgruening) [inline tool_collections/samtools/samtools_view/samtools_view.xml:949]:
```suggestion
            <output name="outputsam" file="test_32.bam" ftype="bam" lines_diff="2" ftype="bam" />
```

**Review by bgruening — APPROVED** [review]
Great! Just a few tiny suggestions. Maybe we should remove quotes from the allowed characters ...?

**Review by bgruening — APPROVED** [review]

**Pull request merged**